vassal-app/src/main/java/VASSAL/build/module/map/CounterDetailViewer.java

```diff
@@ -469,37 +495,28 @@ protected void drawGraphics(Graphics g, @SuppressWarnings("unused") Point pt, JC
-        final Point snapPt = map.snapTo(mapPt, showTerrainSnappy);
-        final String locationName = map.localizedLocationName(snapPt);
-        showTerrainText.setProperty(BasicPiece.LOCATION_NAME, locationName.equals(Resources.getString("Map.offboard")) ? "" : locationName);
-        showTerrainText.setProperty(BasicPiece.CURRENT_MAP, map.getLocalizedMapName());
-        final Board b = map.findBoard(snapPt);
-        final String boardName = (b == null) ? "" : b.getLocalizedName();
-        showTerrainText.setProperty(BasicPiece.CURRENT_BOARD, boardName);
-        final Zone z = map.findZone(snapPt);
-        final String zone = (z == null) ? "" : z.getLocalizedName();
-        showTerrainText.setProperty(BasicPiece.CURRENT_ZONE, zone);
-
-        final String text = showTerrainText.getLocalizedText(this, "Editor.MouseOverStackViewer.text_below");
-        final int y = dbounds.y + dbounds.height + 10 + extraTextPadding * 2;
-        if (text.length() > 0) {
-          // If we're doing the "stretch the bottom all the way across" thing, then draw our "master box" now.
-          if (combineCounterSummary && stretchWidthSummary) {
-            drawLabel(g, new Point(lastPieceBounds.x - 1, y), pieces.isEmpty() ? text : " ", LabelUtils.CENTER, LabelUtils.CENTER,
-              lastPieceBounds.width + 2,
-              lastPieceBounds.width + 2,
-              1,
-              false);
-            didadukey = true;
-          }
-          else {
-            final int x = dbounds.x + (int) (borderOffset * os_scale);
-            drawLabel(g, new Point(x, y), text, LabelUtils.CENTER, LabelUtils.CENTER, (int)(showTerrainWidth * graphicsZoom * os_scale), 0, 0, false);
-          }
-          anyUnderText = true;
+      if (textVisible && !terrainText.isEmpty()) {
+        int y = dbounds.y + dbounds.height + (int)((verticalBottomText + extraTextPadding * 2) * os_scale);
+        if ((useInnerBottom ? borderInnerThickness : borderThickness) > 2) {
+          y -= ((useInnerBottom ? borderInnerThickness : borderThickness) - 2) * os_scale; // Stuff was previously built around a border thickness of 2, so if it's more we attempt to adjust upwards for better results. Designer can manually adjust.
+        }
+        // If we're doing the "stretch the bottom all the way across" thing, then draw our "master box" now.
+        if (combineCounterSummary && stretchWidthSummary) {
+          drawLabel(g, new Point(lastPieceBounds.x - 1, y), pieces.isEmpty() ? terrainText : " ", LabelUtils.CENTER, LabelUtils.CENTER,
+            lastPieceBounds.width + 2,
+            lastPieceBounds.width + (int)((borderThickness * 2) * os_scale),
+            (int)((borderThickness - 1) * os_scale),
+            false,
+            (int)(((useInnerBottom ? borderInnerThickness : borderThickness) - 1) * os_scale),
+            (int)((borderThickness - 1) * os_scale)
+          );
+          didadukey = true;
```

Comment:
What's this?

Reply:
(And just to clarify, this was already in the earlier PR, the one that dealt with the terrain, you're just noticing it now)

Reply:
Could you maybe name this something better? Nobody will have any idea what it is if anyone looks at it in the future.

Reply:
Oh THAT kind of "what's this"... fixed up